custom_components/waste_collection_schedule/waste_collection_schedule/source/moray_gov_uk.py

```diff
@@ -35,28 +32,26 @@ def __init__(self, id):
         self._id = str(id).zfill(8)
 
     def fetch(self):
-        response = requests.Session().get(
-            f"https://bindayfinder.moray.gov.uk/cal_2024_view.php?id={self._id}"
-        )
-        response.raise_for_status()
+        response = requests.Session().get(f"https://bindayfinder.moray.gov.uk/cal_2024_view.php?id={self._id}")
```

Comment:
- I do not like the hard-coded 2024 in the URL (missed this last time). What about using always the current year (`datetime.now().year`)? Or `try` the current year and use 2024 (or  (`datetime.now().year-1`) if it fails (if they do not change the URL). What do you think?
- you do not need to use `requests.Session().get(` use `requests.get(` instead. The session object is only necessary if you want to make multiple requests and keep the HEADERS and COOKIES. You would then save the session in a dedicated variable to call the requests with this session variable
- I'd prefer passing GET parameter with the params argument (`requests.get("...view.php" params={"id":self._id})`

Reply:
1. Yeah, that's a tough one because we're at their (developers') mercy. They only switched to year-based URLs for 2024 - I have no idea whether they'll stick with it for 2025...
2. Thanks - did not know that and probably because I just copied blindly from another implementation.
3. I'll refactor.

Reply:
Thanks for the work on this. I just downloaded waste_collection_schedule last week and figured it wasn't working for Moray and was tinkering around with it and discovered the old URL https://bindayfinder.moray.gov.uk/cal_2023_view.php? still works giving a list of bins and icons per month instead of the shaded days in the calendar (I think the Council website used to give you a choice of which calendar format you preferred?) Changing the sources code for Moray to the old URL pulled the days down into the calendar. I was looking how to advise about this (very new to GitHub!) but it looks like you have already reconfigured to work with the new format which is great!